demos/custom-fields/field_turtle.js

```diff
@@ -0,0 +1,705 @@
+    value = this.cachedValidatedValue_;
+    value.pattern = value.pattern || CustomFields.FieldTurtle.PATTERNS[0];
+    value.hat = value.hat || CustomFields.FieldTurtle.HATS[0];
+    value.turtleName = value.turtleName || CustomFields.FieldTurtle.NAMES[0];
+  }  // Else the original value is fine.
+
+  // A field constructor should always call its parent constructor, because
+  // that helps keep the code organized and DRY.
+  CustomFields.FieldTurtle.superClass_.constructor.call(
+      this, value, opt_validator);
+  this.size_ = new goog.math.Size(72, 40);
+};
+goog.inherits(CustomFields.FieldTurtle, Blockly.Field);
+
+// This allows the field to be constructed using a JSON block definition.
+CustomFields.FieldTurtle.fromJson = function(options) {
+  // In this case we simply pass the JSON options along to the constructor,
+  // but you can also use this to get message references, and other such things.
+  return new CustomFields.FieldTurtle(
+    options['pattern'],
+    options['hat'],
+    options['turtleName']);
+};
+
+// Since this field is editable we must also define serializable as true
+// (for backwards compatibility reasons serializable is false by default).
+CustomFields.FieldTurtle.prototype.SERIALIZABLE = true;
+
+// The cursor property defines what the mouse will look like when the user
+// hovers over the field. By default the cursor will be whatever
+// .blocklyDraggable's cursor is defined as (vis. grab). Most fields define
+// this property as 'default'.
+CustomFields.FieldTurtle.prototype.CURSOR = 'pointer';
+
+// These are the different options for our turtle. Being declared this way
+// means they are static, and not translatable. If you want to do something
+// similar, but make it translatable you should set up your options like a
+// dropdown field, with language-neutral keys and human-readable values.
+CustomFields.FieldTurtle.PATTERNS =
+    ['Dots', 'Stripes', 'Hexagons'];
+CustomFields.FieldTurtle.HATS =
+    ['Stovepipe', 'Crown', 'Propeller', 'Mask', 'Fedora'];
+CustomFields.FieldTurtle.NAMES =
+    ['Yertle', 'Franklin', 'Crush', 'Leonardo', 'Bowser', 'Squirtle', 'Oogway'];
+
+// Used to keep track of our editor event listeners, so they can be
+// properly disposed of when the field closes. You can keep track of your
+// listeners however you want, just be sure to dispose of them!
+CustomFields.FieldTurtle.prototype.editorListeners_ = [];
+
+// Used to create the DOM of our field.
+CustomFields.FieldTurtle.prototype.initView = function() {
+  // Because we want to have both a borderRect_ (background) and a
+  // textElement_ (text) we can call the super-function. If we only wanted
+  // one or the other, we could call their individual createX functions.
+  CustomFields.FieldTurtle.superClass_.initView.call(this);
+
+  // The field group is created by the abstract field's init_ function. *All
+  // elements* should be children of the fieldGroup_.
+  this.movableGroup_ = Blockly.utils.dom.createSvgElement('g',
```

Comment:
You might want to pull out the svg code into a separate function and put it at the bottom of the file. It's not really important to the concepts so moving it out of the way will make it easier for folks to follow the rest of the code.